File: backend/app/models/api/apps.py
"""Schémas Pydantic pour l'endpoint /v1/apps — menu launcher cross-suite."""
from __future__ import annotations

from pydantic import BaseModel, HttpUrl


class AppEntry(BaseModel):
    """Entrée du menu launcher d'applications."""

    key: str
    label: str
    icon: HttpUrl
    url: HttpUrl


class AppsResponse(BaseModel):
    """Réponse de l'endpoint GET /v1/apps."""

    urls: list[AppEntry]

File: frontend/src/schemas/apps.ts
/**
 * Zod schemas for /v1/apps endpoint.
 * Mirror of backend app/models/api/apps.py.
 */
import { z } from 'zod'

export const AppEntrySchema = z.object({
  key: z.string().min(1),
  label: z.string().min(1),
  icon: z.string().url(),
  url: z.string().url(),
})

export const AppsResponseSchema = z.object({
  urls: z.array(AppEntrySchema),
})

export type AppEntry = z.infer<typeof AppEntrySchema>
export type AppsResponse = z.infer<typeof AppsResponseSchema>
